pennylane/resource/resource.py

```diff
@@ -71,6 +75,35 @@ def _ipython_display_(self):
         print(str(self))
 
 
+class OperationWithResources(Operation):
+    @abstractmethod
+    def resources(self) -> Resources:
+        r"""Compute the resources required for this operation.
+
+        Returns:
+            (:class:`~.Resources`): The resources required by this operation.
+
+        **Examples**
+
+        >>> class CustomOp(OperationWithResources):
+        ...     num_wires = 2
+        ...     def resources(self):
+        ...         return Resources(num_wires=self.num_wires, num_gates=3, depth=2)
+        ...
+        >>> op = CustomOp()
```

Comment:
Could be good to have an example like this in the docstring for `OperationWithResources`

Reply:
👍🏼